examples/hosts/app-integration/external-data/src/start.ts

```diff
@@ -51,12 +51,33 @@ async function start(): Promise<void> {
 		model = createResponse.model;
 
 		id = await createResponse.attach();
+
+		// Hardcoding a taskListId here. A follow up will be to introduce a form
+		// where the user can enter an external taskListId that they want
+		// to import from the external server.
+		model.baseDocument.addTaskList({ externalTaskListId: "task-list-1" });
 	} else {
 		id = location.hash.slice(1);
 		model = await tinyliciousModelLoader.loadExisting(id);
 		showExternalServerView = false;
 	}
 
+	// This block is necessary so that we render after the task list
+	// has instantiated. In a future PR this will go away, replaced
+	// by the registration api call.
+	let taskList: ITaskList | undefined;
+	while (taskList === undefined) {
+		const taskListsChangedP = new Promise<void>((resolve) => {
+			model.baseDocument.once("taskListCollectionChanged", () => {
+				resolve();
+			});
+		});
+		taskList = model.baseDocument.getTaskList("task-list-1");
+		if (taskList === undefined) {
+			await taskListsChangedP;
+		}
+	}
```

Comment:
Preferable here to avoid the loop and just make the awaited promise more specific (wait for exactly the right moment), i.e. something like:
```suggestion
	taskList = model.baseDocument.getTaskList("task-list-1");
	if (taskList === undefined) {
		await new Promise<void>((resolve) => {
			const checkForTaskList = () => {
				if (model.baseDocument.getTaskList("task-list-1") !== undefined) {
					model.baseDocument.off("taskListCollectionChanged", checkForTaskList);
					resolve();
				}
			}
			model.baseDocument.on("taskListCollectionChanged", checkForTaskList);
		});
		taskList = model.baseDocument.getTaskList("task-list-1");
	}
```